src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { ReactElement, useEffect, useMemo, useRef } from 'react';
+import { Range } from 'react-range';
+
+import useTheme from '../../hooks/useTheme';
+import { TestProps } from '../../utils/types';
+import TextField from '../TextField';
+import SliderMark from './components/SliderMark';
+import SliderThumb from './components/SliderThumb';
+import SliderTrack from './components/SliderTrack';
+import { Container, InputContainer, InputsContainer } from './Slider.style';
+import { IMarkProps, IThumbProps, ITrackProps } from 'react-range/lib/types';
+
+type CommonProps = {
+  /** Determines if the Slider is disabled or not */
+  disabled?: boolean;
+  /** Determines if the Slider will show increments along its track */
+  hasIncrements?: boolean;
+  /** Called when a thumb is moved, provides new values */
+  onChange: (values: number[]) => void;
+  /** Called when a change is finished (mouse/touch up, or keyup), provides current values.
+   * Use this event when you have to make for example ajax request with new values. */
+  onFinalChange?: (values: number[]) => void;
+  /** Determines the position of the rendered thumbs and the type of the Slider. 1 value means it's a Selector,
+   * while 2 values mean it's a Range. Defaults to [0, 100] */
+  values: [number] | [number, number];
+};
+
+const STEP = 1;
+export const STEP_WITH_INCREMENTS = 20;
+export const MIN = 0;
+export const MAX = 100;
+
+const Slider: React.FC<CommonProps & TestProps> = ({
+  values = [MIN, MAX],
+  onChange,
+  onFinalChange,
+  hasIncrements = false,
+  disabled = false,
+  dataTestPrefixId,
+}) => {
+  const theme = useTheme();
+
+  const isSelector = useMemo(() => values.length === 1, []);
+
+  const initialValueRef = useRef<[number] | [number, number]>();
+
+  useEffect(() => {
+    initialValueRef.current = values;
```

Comment:
...I know I suggested it this way, but I just realized this can be written even simpler:

```ts
const initialValueRef = useRef<[number] | [number, number]>(props.values);
```

This will only run once, so you don't even need the `useEffect`, and then you can instantly set 

```ts
const isSelector = initialValueRef.current.length === 1;
```